### PR TITLE
GEOMESA-2937 Fix schema management index flag checks

### PR DESCRIPTION
**Pull request by elahrvivaz**
* Split up GeoMesaDataStore test classes
* Remove accidental check-in of application conf file

Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

File: geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/geotools/GeoMesaDataStore.scala [hunk 1]
```diff
@@ -547,9 +547,12 @@ abstract class GeoMesaDataStore[DS <: GeoMesaDataStore[DS]](val config: GeoMesaD
     }
     sft.getAttributeDescriptors.asScala.foreach { d =>
       if (indexed(d)) {
-        val fields = Seq(d.getLocalName) ++ Option(sft.getGeomField) ++ sft.getDtgField
-        val existing = sft.getIndices
-        if (!existing.exists(e => e.name == AttributeIndex.name && e.attributes.map(remapCol) == fields)) {
+        val existing = {
```
Comment (elahrvivaz) [inline geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/geotools/GeoMesaDataStore.scala:550]: this is the only logic change

File: geomesa-index-api/src/test/scala/org/locationtech/geomesa/index/geotools/GeoMesaDataStoreAlterSchemaTest.scala [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+      addAttribute must beAnInstanceOf[SchemaCompatibility.Compatible]
+      addAttribute.apply()
+
+      val updateAddAttribute = ds.getSchema("test")
+      updateAddAttribute.getAttributeDescriptors.asScala.map(_.getLocalName) mustEqual
+          Seq("type", "dtg", "geom", "name")
+
+      ds.checkSchemaCompatibility("test", toSft(addAttributeConfig)) mustEqual SchemaCompatibility.Unchanged
+    }
+    "update compatible schemas from index flags" in {
```
Comment (elahrvivaz) [inline geomesa-index-api/src/test/scala/org/locationtech/geomesa/index/geotools/GeoMesaDataStoreAlterSchemaTest.scala:185]: this is the unit test that shows the issue that is fixed

File: geomesa-tools/conf/application2.conf [hunk 3]
```diff
@@ -1,35 +0,0 @@
-geomesa {
```
Comment (elahrvivaz) [inline geomesa-tools/conf/application2.conf:1]: this was committed accidentally a while ago

**Comment by elahrvivaz** [comment]
except where noted, the other changes are just splitting out the different data store tests into 3 files

File: geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/geotools/GeoMesaDataStore.scala (hunk 1 above)
Comment (jnh5y) [reply geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/geotools/GeoMesaDataStore.scala:550]: Were updates removing indexes? :grimacing:

File: geomesa-index-api/src/test/scala/org/locationtech/geomesa/index/geotools/GeoMesaDataStoreVisitorTest.scala [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+ * Copyright (c) 2013-2020 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.index.geotools
+
+
```
Comment (jnh5y) [inline geomesa-index-api/src/test/scala/org/locationtech/geomesa/index/geotools/GeoMesaDataStoreVisitorTest.scala:11]: Extra space!;)

**Review by jnh5y — APPROVED** [review]

File: geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/geotools/GeoMesaDataStore.scala (hunk 1 above)
Comment (elahrvivaz) [reply geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/geotools/GeoMesaDataStore.scala:550]: :sweat_smile:

**Pull request merged**